src/neo/SmartContract/Manifest/ContractEventDescriptor.cs

```diff
@@ -32,6 +32,11 @@ public class ContractEventDescriptor : IInteroperable
         /// </summary>
         public ContractParameterDefinition[] Parameters { get; set; }
 
+        /// <summary>
+        /// When the Method start to be available
+        /// </summary>
+        public uint AvailableFromVersion { get; set; } = 0;
```

Comment:
And this?

Reply:
We need to send the `Attribute` value to the `ContractEventDescriptor` in order to know it for filtering

Reply:
I think we need to update the states.

Reply:
Like https://github.com/neo-project/neo/pull/2632/files#r751093908 ?